### DISCUSSION TITLE
UI changes proposal 9/12/2024

### DISCUSSION THREAD

**Discussion by dylanmartin** [trendscenter/coinstac-mint#4] [General]
Below are some proposed changes to the UI we can discuss. They can be broken out into separate discussions if need be.

### Runs and Updates
- **Remove the runs list** from the consortium page and replace it with a link to a dedicated page with a filtered run list:
  - Filtered by the current consortium.
  - The current user’s `userid` must be in the consortium’s members array.
- **Allow users to reload** the run list page.
- **Allow users to reload** the content of the run details page.
- **Run event notifications** can contain links to the respective run and consortium pages.

### Live Updates in the Consortium Page:
- The UI should **automatically update** when there are changes to the consortium.
- **Consortium information**:
  - Consortium name and description.
  - Members list.
- **Study configuration**:
  - Leader notes.
  - `parameters.json`.
  - Selected computation.

### Study Configuration/Computation UI:
- **Include image download URL**, with options for:
  - `docker pull`.
  - `singularity pull`.
- Change "Settings" label to "parameters.json"

### Miscellaneous UI:
- **Responsive design**:
  - Design for **800x600** resolution first, then adapt to larger screen sizes.
  - Use a **single column layout** when the width is below a defined threshold.
- **Sticky header**: Make the header sticky so it follows the view when scrolling.
- **Clear button functions**: Ensure button actions are clear by providing tooltips.
- **Live state changes**: Ensure live state changes do not interrupt the user when editing a field.

**Comment by pixelsaurus** [comment]
Regarding Responsive Design: 

I don't think it makes sense to design for a single column view since we are not designing for mobile. 800 x 600 has historically accommodated 2 columns.

I think it is useful to show - at a glance - important (if abbreviated) panels in this 2-column structure based on screen size. 

It also bears considering that the 800 x 600 size is in itself in regard to a pre-anticipated edge case where a given user might be using a low resolution virtual machine. In that regard, the default resolution for a VMWare virtual machine is: 1024 x 768. 

We can certainly consider how and what should be displayed between a minimal size and the avg. common viewport size. 

Below are the most common desktop viewport sizes as of 2024. The minimum size here is 1280 x 720, which is close to the current configured default setting of 1200 x 1000. 

- 1920 x 1080
- 1366 x 768
- 1536 x 864
- 1280 x 720
- 1440 x 900
- 1600 x 900

Source: https://www.browserstack.com/guide/common-screen-resolutions

**Reply by dylanmartin** [reply]
I agree that 800x600 isn’t a common use case for most users, and I didn’t mean to suggest it as a hard requirement. Instead, I see it as a useful benchmark to validate and guide our design. Designing with smaller constraints forces us to make thoughtful decisions about UI flow and layout. If we design well, support for 800x600 may come naturally, without much extra effort.

**Reply by pixelsaurus** [reply]
If 800x600 is the self-imposed benchmark by which all design stems, with the assumption that larger screens will then work naturally, then it should be treated as hard design requirement moving forward.

**Reply by dylanmartin** [reply]
I’m not sure that designing for 800x600 should be a permanent hard rule, but it can definitely serve as a useful initial set of constraints to guide us. If we encounter challenges fitting certain features or functions within those constraints, it gives us an opportunity to reflect on why. Maybe the constraints are too narrow or maybe our design could improve.

**Reply by dylanmartin** [reply]
An additional important consideration is that users may want to run the application in windowed mode. They often need to view other windows simultaneously for tasks like:

- Running terminal commands
- Browsing directories
- Monitoring Docker processes

Designing the application to work at less than full screen allows users to multitask and manage multiple processes more effectively.

**Reply by dylanmartin** [reply]
I think using flexbox well would give us all resolutions for free when it comes to number of columns

**Reply by dylanmartin** [reply]
something like this
![flex](https://github.com/user-attachments/assets/7de45b7b-654d-45b2-b454-3bb85a20741b)
```tsx
import React from 'react';
import { Box, Typography, Paper } from '@mui/material';

const ConsortiumPage: React.FC = () => {
  const content = [
    {
      title: "Consortium Members",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec odio. Praesent libero. Sed cursus ante dapibus diam.",
    },
    {
      title: "Data directory",
      description: "Sed nisi. Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta.",
    },
    {
      title: "Consortium Leader Notes",
      description: "Maecenas faucibus mollis interdum. Aenean lacinia bibendum nulla sed consectetur. Cras mattis consectetur purus sit amet fermentum.",
    },
    {
      title: "parameters.json",
      description: "Donec sed odio dui. Curabitur blandit tempus porttitor. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh.",
    },
    {
      title: "Computation Notes",
      description: "Etiam porta sem malesuada magna mollis euismod. Nulla vitae elit libero, a pharetra augue.",
    },
  ];

  return (
    <Box sx={{ padding: 2 }}>
      <Typography variant="h4" gutterBottom>
        Consortium
      </Typography>
      <Box 
        display="grid"
        gridTemplateColumns="repeat(auto-fit, minmax(300px, 1fr))"
        gap={2}
        sx={{ width: '100%', margin: '0 auto' }}
      >
        {content.map((item, index) => (
          <Paper key={index} style={{ padding: 20 }}>
            <Typography variant="h6" gutterBottom>{item.title}</Typography>
            <Typography variant="body1">{item.description}</Typography>
          </Paper>
        ))}
      </Box>
    </Box>
  );
};

export default ConsortiumPage;
```

**Comment by spanta28** [comment]
@pixelsaurus The worst case scenario is users condensing the coinstac screen size to have 2 columns with multiple other windows. So go ahead with that layout. We should not make the app go below 2 columns for the sake of clarity and usability of the app itself. If flexbox is a easy to plugin css tool to use , please go ahead use it otherwise proceed with your code to accommodate the 2 columns.

**Comment by pixelsaurus** [comment]
I have 20+ years of html/css experience. I’m fully aware of responsive practices including flex-box. I’ll take it into consideration, thx.

**Reply by spanta28** [reply]
For sure..Second that!
